src/datatable.js

```diff
@@ -0,0 +1,241 @@
+import {bindable, inject, computedFrom, customElement, bindingMode} from 'aurelia-framework';
+import {resolvedView} from 'aurelia-view-manager';
+import {Router} from 'aurelia-router';
+import {Statham} from 'json-statham';
+
+@customElement('data-table')
+@resolvedView('aurelia-data-table', 'datatable')
+@inject(Router, Element)
+export class DataTable {
+  // search criteria
+  @bindable({defaultBindingMode: bindingMode.twoWay}) criteria;
+
+  // repository resource
+  @bindable repository;
+
+  // amount of items to show on 1 page
+  @bindable limit = 30;
+
+  // String representing the column names
+  @bindable columns = '';
+
+  // Column used by default for search
+  @bindable defaultColumn;
+
+  // Show the search field? (Optional attribute)
+  @bindable searchable = null;
+
+  // Columns can be sorted? (Optional attribute)
+  @bindable sortable = null;
+
+  // Rows are editable? (Optional attribute)
+  @bindable update = null;
+
+  // Rows are removable? (Optional attribute)
+  @bindable destroy = null;
+
+  // Rows are selectable? (Optional attribute)
+  @bindable select;
+
+  @bindable data;
+  @bindable route;
+  @bindable page = 1;
+  @bindable criteriaPager;
+
+  count           = 0;
+  columnsArray    = [];
+  sortingCriteria = {};
+  searchCriteria  = {}
+
+  constructor(Router, element) {
+    this.router  = Router;
+    this.element = element;
+  }
+
+  attached() {
+    this.load();
+  }
+
+  load() {
+    this.criteria = this.buildCriteria();
+
+    this.repository.find(this.criteria, true).then(result => {
+     this.data = result;
+    })
+    .catch(error => {
```

Comment:
Move one line up (chain directly)

Note: This only applies to promises with a single .then() call.